spec/src/main/asciidoc/repository.asciidoc

```diff
@@ -853,7 +854,7 @@ CompletionStage<Void> insertBook(Book book);
 
 
 An _asynchronous repository_ is a repository which declares asynchronous repository methods.
-A Jakarta Data provider is not required to support mixing synchronous and asynchronous repository methods within the same repository interface.
+A mixture of synchronous and asynchronous repository methods can be used within the same repository if all of the asynchronous methods are annotated with the `@Asynchronous` annotation to have the Jakarta Concurrency provider handle the asynchronous behavior. Otherwise, the Jakarta Data provider is not required to support mixing synchronous and asynchronous repository methods within the same repository interface.
```

Comment:
I would like to say something here which allows _reactive_ repositories to _not_ have to support `@Asynchronous`, because it would be some sort of weird noop for them.

Reply:
Yes, that makes sense.  How about this wording?  https://github.com/jakartaee/data/pull/1003/commits/818dfd8c98414fc7029a9e01b609e1889ce8fa90  It disallows `@Asynchronous` in that case.